coalib/misc/CachingUtilities.py

```diff
@@ -136,7 +137,13 @@ def get_settings_hash(sections):
     """
     settings = []
     for section in sections:
+        orig = copy.deepcopy(sections[section])
```

Comment:
actually orig is the copy and you change then the original, rather do it the other way round, make a copy and change that one ;)